src/core/thread/network_data_local.cpp

```diff
@@ -195,13 +196,33 @@ ThreadError Local::UpdateRloc(BorderRouterTlv &aBorderRouter)
     return kThreadError_None;
 }
 
+bool Local::IsOnMeshPrefixConsistent(void)
+{
+    return (mLeader.ContainsOnMeshPrefixes(*this, mMle.GetRloc16()) &&
+            ContainsOnMeshPrefixes(mLeader, mMle.GetRloc16()));
+}
+
+bool Local::IsExternalRouteConsistent(void)
+{
+    return (mLeader.ContainsExternalRoutes(*this, mMle.GetRloc16()) &&
+            ContainsExternalRoutes(mLeader, mMle.GetRloc16()));
+}
+
 ThreadError Local::SendServerDataNotification(void)
 {
     ThreadError error = kThreadError_None;
     uint16_t rloc = mMle.GetRloc16();
 
```

Comment:
currently in our THCI `configBorderRouter()` API implementation, `prefix` and `netdataregister` would always be bundled together.
we might want to add `VerifyOrExit((mMle.GetDeviceState() != Mle::kDeviceStateDisabled), ;);` to avoid confusing `kThreadError_InvalidState` error status.